internal/subscription/core.go

```diff
@@ -343,54 +342,39 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
 
 	logger.Ctx(ctx).Infow("migration: found subscriptions to migrate", "count", len(subscriptionsToUpdate))
 
-	topicNames := make([]string, 0)
-	for _, model := range subscriptionsToUpdate {
-		needsUpdate := false
-
-		// update retry policy if not set
-		if model.RetryPolicy == nil {
-			model.setDefaultRetryPolicy()
-			// update dead letter policy every time as previously we were saving only dead-letter topic names
-			// this takes care of updating max_delivery_attempts
-			model.setDefaultDeadLetterPolicy()
-			needsUpdate = true
+	updatedSubCount := 0
+	for _, sub := range subscriptionsToUpdate {
+		if sub.PushConfig == nil {
```

Comment:
```suggestion
		if sub.PushConfig == nil || sub.PushConfig.Attributes == nil {
```
no need for `if pushAttr != nil` check below.